File: android-app/app/src/main/java/com/vektra/ehrms/ui/management/KioskScreen.kt
package com.vektra.ehrms.ui.management

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.vektra.ehrms.data.models.FaceCheckInResponse
import com.vektra.ehrms.data.repository.VektraRepository
import com.vektra.ehrms.ui.components.CameraXPreview
import com.vektra.ehrms.ui.theme.*
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@Composable
fun KioskScreen() {
    val repository = remember { VektraRepository() }
    val scope = rememberCoroutineScope()

    var isCapturing by remember { mutableStateOf(false) }
    var isProcessing by remember { mutableStateOf(false) }
    var matchResult by remember { mutableStateOf<FaceCheckInResponse?>(null) }
    var statusText by remember { mutableStateOf("Position your face in front of the camera terminal") }

    fun processKioskFrame(currB64: String, prevB64: String?) {
        isProcessing = true
        statusText = "Analyzing facial biometrics..."

        scope.launch {
            val res = repository.verifyAndCheckInFace(currB64, prevB64)
            isProcessing = false
            isCapturing = false

            res.onSuccess { data ->
                matchResult = data
                statusText = data.message

                // Reset overlay after 4 seconds for next person in line
                delay(4000)
                matchResult = null
                statusText = "Position your face in front of the camera terminal"
            }.onFailure { err ->
                statusText = err.message ?: "Face not recognized. Please face camera directly."
                delay(3000)
                statusText = "Position your face in front of the camera terminal"
            }
        }
    }

    Column(
        modifier = Modifier.fillMaxSize().background(Navy900).padding(16.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column {
                Text("Touchless Biometric Kiosk", fontSize = 22.sp, fontWeight = FontWeight.Bold, color = Color.White)
                Text("High-Speed Counter Terminal (1:N Recognition)", fontSize = 12.sp, color = Slate400)
            }

            Surface(color = Emerald500.copy(alpha = 0.2f), shape = RoundedCornerShape(12.dp)) {
                Text("Terminal Active", modifier = Modifier.padding(8.dp), color = Emerald500, fontSize = 11.sp, fontWeight = FontWeight.Bold)
            }
        }

        Box(
            modifier = Modifier
                .fillMaxWidth()
                .weight(1f)
                .background(Navy800, shape = RoundedCornerShape(24.dp)),
            contentAlignment = Alignment.Center
        ) {
            CameraXPreview(
                modifier = Modifier.fillMaxSize(),
                useFrontCamera = true,
                isCapturing = isCapturing,
                onFrameCaptured = { curr, prev -> processKioskFrame(curr, prev) }
            )

            // Result Match Card Overlay
            matchResult?.let { match ->
                Surface(
                    color = Color.Black.copy(alpha = 0.85f),
                    modifier = Modifier.fillMaxSize()
                ) {
                    Column(
                        modifier = Modifier.fillMaxSize().padding(24.dp),
                        horizontalAlignment = Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.Center
                    ) {
                        Icon(Icons.Default.CheckCircle, contentDescription = null, tint = Emerald500, modifier = Modifier.size(64.dp))
                        Spacer(modifier = Modifier.height(16.dp))
                        Text(match.employee?.name ?: "Employee Matched", fontSize = 24.sp, fontWeight = FontWeight.Bold, color = Color.White)
                        Text("${match.employee?.department} • ID: ${match.employee?.employeeId}", fontSize = 14.sp, color = Slate400)
                        Spacer(modifier = Modifier.height(12.dp))
                        Surface(color = Emerald500, shape = RoundedCornerShape(14.dp)) {
                            Text(
                                text = "${match.actionType ?: "CHECK_IN"} SUCCESS (${match.verification?.confidence ?: 98.0}% Match)",
                                modifier = Modifier.padding(horizontal = 16.dp, vertical = 8.dp),
                                color = Color.White,
                                fontWeight = FontWeight.Bold
                            )
                        }
                    }
                }
            }

            if (isProcessing) {
                Surface(color = Color.Black.copy(alpha = 0.6f), modifier = Modifier.fillMaxSize()) {
                    Box(contentAlignment = Alignment.Center) {
                        CircularProgressIndicator(color = Emerald500)
                    }
                }
            }
        }

        Text(statusText, color = Slate400, fontSize = 13.sp, fontWeight = FontWeight.Medium)

        Button(
            onClick = { isCapturing = true },
            modifier = Modifier.fillMaxWidth().height(52.dp),
            shape = RoundedCornerShape(14.dp),
            colors = ButtonDefaults.buttonColors(containerColor = Emerald500),
            enabled = !isProcessing && !isCapturing && matchResult == null
        ) {
            Icon(Icons.Default.Refresh, contentDescription = null, tint = Color.White)
            Spacer(modifier = Modifier.width(8.dp))
            Text("TRIGGER BIOMETRIC SCAN", fontWeight = FontWeight.Bold)
        }
    }
}

File: android/src/screens/management/KioskScreen.js
import React, { useState, useRef } from 'react';
import { View, Text, StyleSheet, TouchableOpacity, ActivityIndicator } from 'react-native';
import { CameraView } from 'expo-camera';
import { Ionicons } from '@expo/vector-icons';
import Header from '../../components/Header';
import api from '../../api/axios';
import { COLORS } from '../../theme/colors';

export default function KioskScreen() {
  const cameraRef = useRef(null);
  const [capturing, setCapturing] = useState(false);
  const [matchResult, setMatchResult] = useState(null);
  const [statusText, setStatusText] = useState('Position your face in front of the camera terminal');

  const handleKioskScan = async () => {
    if (!cameraRef.current) return;

    setCapturing(true);
    setStatusText('Analyzing facial biometrics...');
    setMatchResult(null);

    try {
      const photo1 = await cameraRef.current.takePictureAsync({ base64: true, quality: 0.6 });
      await new Promise(r => setTimeout(r, 250));
      const photo2 = await cameraRef.current.takePictureAsync({ base64: true, quality: 0.6 });

      const res = await api.post('face/verify-checkin', {
        image: `data:image/jpeg;base64,${photo2.base64}`,
        prevImage: `data:image/jpeg;base64,${photo1.base64}`,
      });

      setCapturing(false);
      setMatchResult(res.data);
      setStatusText(res.data?.message || 'Verification complete!');

      setTimeout(() => {
        setMatchResult(null);
        setStatusText('Position your face in front of the camera terminal');
      }, 4000);
    } catch (err) {
      setCapturing(false);
      const msg = err.response?.data?.message || 'Face not recognized. Please face camera directly.';
      setStatusText(msg);
      setTimeout(() => setStatusText('Position your face in front of the camera terminal'), 3000);
    }
  };

  return (
    <View style={styles.container}>
      <Header title="Touchless Kiosk Mode" />
      <View style={styles.content}>
        <Text style={styles.subtitle}>High-Speed Reception Counter Terminal (1:N Match)</Text>

        <View style={styles.cameraBox}>
          <CameraView style={styles.camera} facing="front" ref={cameraRef} />

          {matchResult && (
            <View style={styles.matchOverlay}>
              <Ionicons name="checkmark-circle-outline" size={64} color={COLORS.emerald500} />
              <Text style={styles.matchName}>{matchResult.employee?.name || 'Matched'}</Text>
              <Text style={styles.matchSub}>{matchResult.employee?.department} • ID: {matchResult.employee?.employeeId}</Text>
              <View style={styles.matchTag}>
                <Text style={styles.matchTagText}>
                  {matchResult.actionType || 'CHECK_IN'} SUCCESS ({matchResult.verification?.confidence || 98}% Match)
                </Text>
              </View>
            </View>
          )}

          {capturing && (
            <View style={styles.loadingOverlay}>
              <ActivityIndicator size="large" color={COLORS.emerald500} />
            </View>
          )}
        </View>

        <Text style={styles.statusText}>{statusText}</Text>

        <TouchableOpacity
          style={styles.scanBtn}
          onPress={handleKioskScan}
          disabled={capturing || !!matchResult}
        >
          <Ionicons name="refresh" size={20} color="#FFF" style={{ marginRight: 8 }} />
          <Text style={styles.scanBtnText}>TRIGGER BIOMETRIC SCAN</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: COLORS.screenBg },
  content: { flex: 1, padding: 16, alignItems: 'center' },
  subtitle: { color: COLORS.slate400, fontSize: 12, marginBottom: 16 },
  cameraBox: { width: '100%', flex: 1, borderRadius: 24, overflow: 'hidden', backgroundColor: COLORS.cardBg, marginBottom: 16 },
  camera: { flex: 1 },
  loadingOverlay: { ...StyleSheet.absoluteFillObject, backgroundColor: 'rgba(0,0,0,0.6)', justifyContent: 'center', alignItems: 'center' },
  matchOverlay: { ...StyleSheet.absoluteFillObject, backgroundColor: 'rgba(15, 23, 42, 0.95)', justifyContent: 'center', alignItems: 'center', padding: 20 },
  matchName: { color: '#FFF', fontWeight: 'bold', fontSize: 22, marginTop: 10 },
  matchSub: { color: COLORS.slate400, fontSize: 13, marginTop: 2 },
  matchTag: { backgroundColor: COLORS.emerald500, paddingHorizontal: 16, paddingVertical: 8, borderRadius: 12, marginTop: 16 },
  matchTagText: { color: '#FFF', fontWeight: 'bold', fontSize: 12 },
  statusText: { color: COLORS.slate400, fontSize: 13, marginBottom: 16, textAlign: 'center' },
  scanBtn: { width: '100%', height: 52, backgroundColor: COLORS.emerald500, borderRadius: 14, flexDirection: 'row', justifyContent: 'center', alignItems: 'center' },
  scanBtnText: { color: '#FFF', fontWeight: 'bold', fontSize: 14 },
});
